db/migrate/20240721115247_add_category_id_to_products.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddCategoryIdToProducts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :products, :category, foreign_key: true, type: :uuid
+    change_column_null :products, :category_id, false
```

Comment:
Set after resolving

Reply:
why we have to add 2 lines whereas 1 line can solve, the second line is redundant

``` ruby
    add_reference :products, :category, foreign_key: true, type: :uuid, null: false, index {unique: true}
  ```

Reply:
Done after sugggestion